torch/csrc/distributed/c10d/init.cpp

```diff
@@ -106,6 +106,26 @@ class PythonStore : public ::c10d::Store {
   }
 };
 
+// PythonhookBinder has a static function called register_comm_hook
+// that enables registering a c10d PythonCommHook object to c10d
+// reducer from Python.
+class PythonHookBinder {
+ public:
+  // This static method is called from DDP's Python API. Its inputs are
+  // a c10d reducer object, state, and callable comm_hook. State and
+  // comm_hook inputs are Python objects and this function creates a
+  // c10d PythonCommHook object using these inputs. It later calls
+  // register_comm_hook function of the reducer input to register that
+  // PythonCommHook object.
+  static void register_comm_hook(
+      ::c10d::Reducer& reducer,
```

Comment:
I think it is better to pass DDP wrapped model here, then inside register_comm_hook , you can get model.reducer(), as reducer usually has dependency on ddp wrapped model.

Reply:
Makes sense. I've changed that. `test_ddp_comm_hook_register_just_once` unit test now depends on DDP model, so I moved that unit test too.